File: schemas/clusters/publication_cluster.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

class PublicationClusterSchemaBase(BaseModel):
    publication_cluster_id: int
    publication_id: int
    cluster_id: int
    cluster_id_string: str
    sdg: Optional[int]
    level: Optional[int]
    topic: Optional[int]

    model_config = {
        "from_attributes": True
    }

class PublicationClusterSchemaFull(PublicationClusterSchemaBase):
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend2/types/clusters/publicationCluster.ts
export interface PublicationClusterSchemaBase {
  publicationClusterId: number;
  publicationId: number;
  clusterId: number;
  clusterIdString: string;
  sdg?: number | null;
  level?: number | null;
  topic?: number | null;
}

export interface PublicationClusterSchemaFull extends PublicationClusterSchemaBase {
  createdAt: string;
  updatedAt: string;
}
